### PR TITLE
Development

### PR DESCRIPTION
**Pull request by FlyingBird95**
Implemented outlier-detection funcitonallity

### CODE REVIEW

File: dashboard/config.py [hunk 1]
```diff
@@ -41,7 +42,11 @@ def from_file(self, config_file):
             USERNAME: for logging into the dashboard, a username and password is required. The
                 username can be set using this variable.
             PASSWORD: same as for the username, but this is the password variable.
-            
+
+            OUTLIER_DETECTION_CONSTANT: When the execution time is more than this constant above
```
Comment (mircealungu) [inline dashboard/config.py]: more than "average * this constant" would be the correction version i'd say.

File: dashboard/config.py [hunk 2]
```diff
@@ -41,7 +42,11 @@ def from_file(self, config_file):
...
+                average, extra information is logged into the database. A default value for this
+                variable is 2.5, but can be changed to a different value.
```
Comment (mircealungu) [inline dashboard/config.py]: but can be changed in the config file

File: dashboard/routings/result.py [hunk 3]
```diff
@@ -56,6 +57,11 @@ def result(end, index=0):
         page = 'endpoint/endpoint_plotly.html'
         graph = get_time_per_user(end)
 
+    # returns a page with a table containing the outliers.
+    elif index == 7:
```
Comment (mircealungu) [inline dashboard/routings/result.py]: i'm not sure of these magic numbers in the code. 
can't we define symbolic constants for them?

**Comment by FlyingBird95** [comment]
Fixed all comments you've made.
Should be ready for merging now

File: dashboard/routings/measurements.py [hunk 4]
```diff
@@ -2,77 +2,123 @@
...
+            marker=dict(
+                color=get_color(endpoint)
+            )
+        ))
 
-    return graph.render_data_uri()
+    layout = go.Layout(
+        barmode='stack',
+        autosize=True,
+        height=350 + 40 * len(labels),
```
Comment (mircealungu) [inline dashboard/routings/measurements.py]: magic constants in the code :)

File: dashboard/routings/measurements.py [hunk 5]
```diff
@@ -108,21 +164,31 @@ def get_boxplot_per_endpoint():
...
+
+    if len(data) == 0:
+        return None
 
     layout = go.Layout(
-        autosize=False,
-        width=900,
+        autosize=True,
         height=350 + 40 * len(endpoints),
         plot_bgcolor='rgba(249,249,249,1)',
```
Comment (mircealungu) [inline dashboard/routings/measurements.py]: shouldn't this rgba(249,249,...) be declared a constant, I see it repeated quite often...

File: dashboard/security.py [hunk 6]
```diff
@@ -1,36 +1,65 @@
...
     def wrapper(*args, **kwargs):
         if session and session.get(config.link + '_logged_in'):
             return func(*args, **kwargs)
-        else:
-            return redirect(url_for('dashboard.login'))
+        return redirect(url_for('dashboard.login'))
     return wrapper
 
 
+def security(admin_only=False):
```
Comment (mircealungu) [inline dashboard/security.py]: err. what's this thing doing?

File: setup.py [hunk 7]
```diff
@@ -1,20 +1,22 @@
...
+    description="A dashboard for automatic monitoring of python web-services",
+    install_requires=['flask>=0.9',         # for monitoring the web-service
+                      'sqlalchemy>=1.1.9',  # for database support
+                      'wtforms>=2.1',       # for generating forms
+                      'flask_wtf',          # also for generating forms
+                      'plotly',             # for generating graphs
+                      'configparser',       # for parsing the config-file
+                      'psutil',             # for logging extra CPU-info
+                      'colorhash',          # for hashing a string into a color
+                      'requests']           # for submitting unit test results
```
Comment (mircealungu) [inline setup.py]: i like that you have comments for the requires!!

**Comment by mircealungu** [comment]
I've merged because my comments were quite minor, but I hope you'll address them nevertheless :)

**Pull request merged**